### PR TITLE
[sc-39654] Add ghciwatch

### PR DESCRIPTION
**Pull request by ysangkok**
Tested locally. I moved down the `ADD` such that it becomes easier to debug. When the ADD is at the top, you have to reinstall GHC and HLS every time you make a change to `install-tools.sh`.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -45,4 +43,6 @@ ARG CABAL_VERSION
 RUN ghcup install cabal $CABAL_VERSION --set
 
 ADD stack.yaml /stack.yaml
-RUN /bin/sh /install-tools.sh
+
+ADD install-tools.sh /install-tools.sh
+RUN bash /install-tools.sh
```
Comment (telser) [inline Dockerfile:48]: Please don't use base. We don't need it for the tests in the install-tools.sh anyway

File: install-tools.sh [hunk 2]
```diff
@@ -1,5 +1,16 @@
 set -o errexit
 
+arch=$(uname -m)
+if [[ $arch == x86_64 ]]; then
```
Comment (telser) [inline install-tools.sh:4]: We should be able to do this with a normal call to `test` in bourne shell. ShellCheck is a big help for this kind of thing.

Comment (ysangkok) [reply install-tools.sh]: Addressed in [66763f3](https://github.com/flipstone/haskell-tools/pull/10/commits/66763f3defcdb74d034bde97f1452266fa6b0a8a)

File: Dockerfile (hunk 1 above)
Comment (ysangkok) [reply Dockerfile]: Addressed in [66763f3](https://github.com/flipstone/haskell-tools/pull/10/commits/66763f3defcdb74d034bde97f1452266fa6b0a8a)

**Review by telser — APPROVED** [review]

**Pull request merged**